kafkatest/broker.go

```diff
@@ -183,7 +184,7 @@ func (p *Producer) Produce(topic string, partition int32, messages ...*proto.Mes
 	if p.ResponseError != nil {
 		return 0, p.ResponseError
 	}
-	off := p.ResponseOffset
+	off := atomic.AddInt64(&p.ResponseOffset, int64(len(messages)))
```

Comment:
`AddInt64` returns the new value.  I think we need to subtract `len(messages)` from `off` before continuing here.